sts_synindex_external.R

```diff
@@ -78,83 +70,125 @@ copy_folders_reparent <- function(source_folder, destination_folder) {
-  dplyr::mutate(file_key = stringr::str_sub(string = path, start = STR_LEN_PARQUET_FINAL_LOCATION+2)) %>% # location of file from home folder of S3 bucket
-  dplyr::mutate(s3_file_key = paste0(file_key)) %>% # the namespace for files in the S3 bucket is S3::bucket/main/
-  dplyr::mutate(md5_hash = as.character(tools::md5sum(path))) %>% 
-  dplyr::ungroup()
-
-## All currently indexed files in Synapse
-synapse_fileview <- synapser::synTableQuery(paste0('SELECT * FROM ', SYNAPSE_FILEVIEW_ID))$filepath %>% read.csv()
-
-## find those files that are not in the fileview - files that need to be indexed
-if (nrow(synapse_fileview)>0) {
-  synapse_manifest_to_upload <- 
-    synapse_manifest %>% 
-    dplyr::anti_join(
-      synapse_fileview %>% 
-        dplyr::select(parent = parentId,
-                      s3_file_key = dataFileKey,
-                      md5_hash = dataFileMD5Hex))
-  } else {
-    synapse_manifest_to_upload <- synapse_manifest
-  }
 
-## Index in Synapse
-## For each file index it in Synapse given a parent synapse folder
-if(nrow(synapse_manifest_to_upload) > 0){ # there are some files to upload
-  for(file_number in seq(nrow(synapse_manifest_to_upload))){
+# De-identify parquet datasets --------------------------------------------
+source('~/recover-parquet-external/deidentification.R')
 
-    # file and related synapse parent id
-    file_= synapse_manifest_to_upload$path[file_number]
-    parent_id = synapse_manifest_to_upload$parent[file_number]
-    s3_file_key = synapse_manifest_to_upload$s3_file_key[file_number]
-    # this would be the location of the file in the S3 bucket, in the local it is at {AWS_PARQUET_DOWNLOAD_LOCATION}
+# Sync final parquets to bucket -------------------------------------------
 
-    absolute_file_path <- tools::file_path_as_absolute(file_) # local absolute path
+date <- lubridate::today()
+sync_cmd <- glue::glue('aws s3 --profile service-catalog sync {PARQUET_FINAL_LOCATION} {base_s3_uri_archive}{date}/ --exclude "*owner.txt*" --exclude "*archive*"')
+system(sync_cmd)
 
-    temp_syn_obj <- synapser::synCreateExternalS3FileHandle(
-      bucket_name = "recover-main-project",
-      s3_file_key = s3_file_key, #
-      file_path = absolute_file_path,
-      parent = parent_id
-    )
 
-    # synapse does not accept ':' (colon) in filenames, so replacing it with '_colon_'
-    new_fileName <- stringr::str_replace_all(temp_syn_obj$fileName, ':', '_colon_')
+# Upload parquet datasets directory tree to Synapse ------------------------
 
-    f <- File(dataFileHandleId=temp_syn_obj$id,
-              parentId=parent_id,
-              name = new_fileName) ## set the new file name
+existing_dirs <- synGetChildren(PARQUET_FOLDER_ARCHIVE) %>% as.list()
 
-    f <- synStore(f)
+if(length(existing_dirs)>0) {
+  for (i in seq_along(existing_dirs)) {
+    synDelete(existing_dirs[[i]]$id)
+  }
+}
 
+# Modify cohort identifier in dir name
+replace_equal_with_underscore <- function(directory_path) {
```

Comment:
Nit: Move all functions to the top so code is more readable.

Reply:
Or move all functions into a file and source it at the top